pump_control.py

```diff
@@ -0,0 +1,57 @@
+"""
+Module to manage a pump with optional PWM functionality.
+The pump will turn on for a specified interval upon calling start_pump.
+Set Interval to 0 to disable this functionality.
+
+Usage Example:
+
+pump = PumpControl(p_pwm_id='P20', p_channel_enable_id='P21', duty_cycle=0.5, interval=10, channel_name="cactus")
+pump.start_pump()
+
+"""
+
+
+from machine import Pin
+from machine import PWM
+from machine import Timer
+from tb6612_single_channel import TB6612_1CH
+
+class PumpControl(TB6612_1CH):
+
+    def __init__(self, p_pwm_id, p_channel_enable_id, channel_name, pwm_timer_id=0, pwm_channel_id=0, duty_cycle=1.0, interval=5):
+        """
+        p_pwm_id            -- the id of the pwm pin (e.g, connect pin to PWMA)
+        p_channel_enable_id -- the id of the motor enable pin (e.g., connect pin to AIN2)
+        channel_name        -- the name of the channel used for logging
+        pwm_timer_id        -- [0], id of the timer, must be 0 to 3
+        pwm_channel_id      -- [0], id of the pwm channel, must be 0 to 7
+        """
+        super().__init__(p_pwm_id, p_channel_enable_id, channel_name, pwm_timer_id, pwm_channel_id)
+        self.__duty_cycle = duty_cycle
+        self.__interval = interval
```

Comment:
is this the watering time? can you add a comment above?

Reply:
Yes. It's sort of in the module description. I can add that.